src/metrics/loc.rs

```diff
@@ -1428,4 +1477,284 @@ mod tests {
+             }",
+            "foo.java",
+            JavaParser,
+            loc,
+            [
+                (lloc, 4, usize), // The number of statements is 4
+            ]
+        );
+    }
+
+    #[test]
+    fn java_multi_lloc() {
+        check_metrics!(
+            "int max = 10; // +1
+
+            for (int i = 0; i < max; i++) { // +3
+               System.out.println(i); // +1
+             }",
+            "foo.java",
+            JavaParser,
+            loc,
+            [
+                (lloc, 5, usize), // The number of statements is 5
+            ]
+        );
+    }
+
+    #[test]
+    fn java_class_loc() {
+        check_metrics!(
+            "
+            public class Person {
+              private String name;
+              public Person(String name){
+                this.name = name; // +1
+              }
+              public String getName() {
+                return name; // +1
+              }
+            }",
+            "foo.java",
+            JavaParser,
+            loc,
+            [
+                (lloc, 2, usize), // The number of statements is 2
+            ]
+        );
+    }
+
+    #[test]
+    fn java_statement_inline_loc() {
+        check_metrics!(
+            "for (int i = 0; i < 100; i++) { System.out.println(\"hello\"); }",
+            "foo.java",
+            JavaParser,
+            loc,
+            [
+                (ploc, 1, usize), // The number of code lines is 1
+                (lloc, 4, usize), // The number of statements is 4
+                (cloc, 0, usize), // The number of comments is 0
+            ]
+        );
+    }
+
+    #[test]
+    fn java_general_loc() {
+        check_metrics!(
+            "int max = 100;
+            
+            /*
+              Loop through and print
+                from: 0
+                to: max
+            */
+            for (int i = 0; i < max; i++) {
+               // Print the value
+               System.out.println(i);
+             }",
+            "foo.java",
+            JavaParser,
+            loc,
+            [
+                (sloc, 11, usize), // The number of lines is 11
+                (ploc, 4, usize),  // The number of code lines is 4
+                (lloc, 5, usize),  // The number of statements is 5
+                (cloc, 6, usize),  // The number of comments is 6
+                (blank, 1, usize)  // The number of blank lines is 1
+            ]
+        );
+    }
+
+    #[test]
+    fn java_main_class_loc() {
+        check_metrics!(
+            "package com.company;
+             /**
+             * The HelloWorldApp class implements an application that
+             * simply prints \"Hello World!\" to standard output.
+             */
+            
+            class HelloWorldApp {
+              public void main(String[] args) {
+                String message = args.length == 0 ? \"Hello empty world\" : \"Hello world\"; // +2 lloc : 1 var assignment + binary exp
```

Comment:
yes it does.

Reply:
So after reading lloc doc on wikipedia: we should have 1 here: for all the assignment, which should be an ExpressionStatement.

Reply:
@calixteman I missed this comment but I had a similar thought. I went with `ExpressionStatement` instead of method invocation and a few other things like handling `i++;`.
This specific line doesn't parse as an expression statement but does count as 1 LLOC as it is a `LocalVariableDeclaration`.